packages/forma-36-react-components/src/components/Notification/NotificationsManager.tsx

```diff
@@ -120,7 +120,9 @@ export class NotificationsManager extends PureComponent<
 
   show: ShowAction<Notification> = (text, settings) => {
     const duration =
-      settings && settings.duration ? settings.duration : this.state.duration;
+      settings && Number.isInteger(settings.duration)
```

Comment:
Failing Typescript linting:

```
Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.

123       settings && Number.isInteger(settings.duration)
                                       ~~~~~~~~~~~~~~~~~
```

Reply:
It's failing because the function `Number.isInteger` can only take a number as argument. Since you above said that `duration` can also be `undefined` the compiler complaints about passing it to `Number.isInteger`.  You'd have to do something like:

```typescript
// The ? after settings in the optional chaining operator
// http://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining 
const duration = Number.isInteger(settings?.duration || this.state.duration)

```